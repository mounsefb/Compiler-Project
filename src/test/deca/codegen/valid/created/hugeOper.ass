	TSTO #6
	BOV stack_overflow_error
	ADDSP #19
; start main program
; Main program
; Beginning of main instructions:
	LOAD #0x1.333334p0, R2
	LOAD #25, R3
	FLOAT R3, R3
	ADD R3, R2
	BOV overflow_error
	LOAD #69, R3
	LOAD #52, R4
	MUL R4, R3
	LOAD #56, R4
	QUO R4, R3
	LOAD #2, R4
	MUL R4, R3
	FLOAT R3, R3
	SUB R3, R2
	BOV overflow_error
	LOAD #65, R3
	FLOAT R3, R3
	ADD R3, R2
	BOV overflow_error
	LOAD #95, R3
	LOAD #52, R4
	MUL R4, R3
	LOAD #85, R4
	QUO R4, R3
	FLOAT R3, R3
	SUB R3, R2
	BOV overflow_error
	LOAD #62, R3
	FLOAT R3, R3
	ADD R3, R2
	BOV overflow_error
	LOAD #95, R3
	LOAD #52, R4
	MUL R4, R3
	LOAD #52, R4
	QUO R4, R3
	LOAD #88, R4
	QUO R4, R3
	FLOAT R3, R3
	SUB R3, R2
	BOV overflow_error
	LOAD #95, R3
	LOAD #56, R4
	MUL R4, R3
	FLOAT R3, R3
	SUB R3, R2
	BOV overflow_error
	LOAD #325, R3
	LOAD #522, R4
	QUO R4, R3
	LOAD #6500, R4
	MUL R4, R3
	FLOAT R3, R3
	ADD R3, R2
	BOV overflow_error
	LOAD #820, R3
	LOAD #100, R4
	MUL R4, R3
	LOAD #20, R4
	QUO R4, R3
	FLOAT R3, R3
	SUB R3, R2
	BOV overflow_error
	STORE R2, 1(GB)
	LOAD 1(GB), R1
	WFLOAT
	WNL
	HALT
; end main program
overflow_error:
	WSTR "Error: Overflow during arithmetic operation"
	WNL
	ERROR
stack_overflow_error:
	WSTR "Error: Stack Overflow "
	WNL
	ERROR
io_error:
	WSTR "Error: Input/Output error"
	WNL
	ERROR
Dereferencement.null:
	WSTR "Erreur : dereferencement de null"
	WNL
	ERROR
pile_pleine:
	WSTR "Erreur : pile pleine"
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : allocation impossible, tas plein"
	WNL
	ERROR
cast_incorrect:
	WSTR "la conversion (cast) est incorrecte"
	WNL
	ERROR
; code de l'initialisation des champs
; code des m√©thodes
code.object.equals:
	LOAD -2(LB), R0
	CMP -3(LB), R0
	BEQ if_equal_pour_object.equals
	LOAD #0, R0
	BRA fin_methodes
if_equal_pour_object.equals:
	LOAD #1, R0
	BRA fin_methodes
fin_methodes:
	RTS
