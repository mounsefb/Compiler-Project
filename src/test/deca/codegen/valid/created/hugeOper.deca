// Description :
// on teste un grand nombre d'oprations 
// RÃ©sultats :
//    -9.45380e+03
{
    float x =  1.2 + 25 - 69 * 52 / 56 * 2 + 65 - 95 * 52 / 85 + 62 - 95 * 52 / 52 / 88 - 95 * 56 + 325 / 522 * 6500 - 820 * 100 / 20;
    println(x);
}