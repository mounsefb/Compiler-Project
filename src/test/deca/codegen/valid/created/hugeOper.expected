-9.45380e+03
